src/panels/lovelace/common/create-badge-element.ts

```diff
@@ -0,0 +1,73 @@
+import deepClone from "deep-clone-simple";
+
+import { fireEvent } from "../../../common/dom/fire_event";
+
+import "../badges/hui-entity-filter-badge";
+import {
+  createErrorBadgeElement,
+  createErrorBadgeConfig,
+  HuiErrorBadge,
+} from "../badges/hui-error-badge";
+import "../badges/hui-state-label-badge";
+import { LovelaceBadge } from "../types";
+import { LovelaceBadgeConfig } from "../../../data/lovelace";
+
+const BADGE_TYPES = new Set(["entity-filter", "error", "state-label"]);
+const CUSTOM_TYPE_PREFIX = "custom:";
+const TIMEOUT = 2000;
+
+const _createElement = (
+  tag: string,
+  config: LovelaceBadgeConfig
+): LovelaceBadge | HuiErrorBadge => {
+  const element = document.createElement(tag) as LovelaceBadge;
+  try {
+    element.setConfig(deepClone(config));
+  } catch (err) {
+    // tslint:disable-next-line
+    console.error(tag, err);
+    return _createErrorElement(err.message);
+  }
+  return element;
+};
+
+const _createErrorElement = (error: string): HuiErrorBadge =>
+  createErrorBadgeElement(createErrorBadgeConfig(error));
+
+export const createBadgeElement = (
+  config: LovelaceBadgeConfig
+): LovelaceBadge | HuiErrorBadge => {
+  if (!config || typeof config !== "object") {
+    return _createErrorElement("No config");
+  }
+
+  if (!config.type) {
+    config.type = "state-label";
```

Comment:
Let's not change the config of the user:
```js
let type = config.type;
if (!type) {
    type = "state-label";
}
```

Reply:
this doesn't flow back to the user, does it?

Reply:
The `config` parameter is passed in, so now you're changing a passed in object.

Reply:
derp, I looked at bram's comment and thought that was my code... :see_no_evil: